Detectors/Base/include/DetectorsBase/ContVec.h

```diff
@@ -0,0 +1,362 @@
+#ifndef O2_BASE_CONTVEC_H
+#define O2_BASE_CONTVEC_H
+
+#include <TTree.h>
+#include <TBranch.h>
+#include <TBranchElement.h>
+#include <TBuffer.h>
+#include <TObject.h>
+#include <memory>
+#include <algorithm>
+#include <type_traits>
+#include <stdexcept>
+#include <iostream>
+#include <FairLogger.h>
+
+
+namespace o2 {
+namespace Base {
+
+using sizeType = int;
+
+ template <class T, class H>
+  class ContVec: public TObject {
+
+ public:
+  
+  struct Header {
+    H         userInfo;         // user assigned data info
```

Comment:
maybe the userInfo could/should go to the metadata and not directly into the payload buffer?